components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/impl/FederatedAuthenticatorConfigBuilderFactory.java

```diff
@@ -141,6 +141,37 @@ public static FederatedAuthenticator build(FederatedAuthenticatorConfig config)
 
         return federatedAuthenticator;
     }
+
+    /**
+     * Builds a list of FederatedAuthenticatorListItem instances based on the given array of
+     * FederatedAuthenticatorConfig.
+     *
+     * @param fedAuthConfigs Array of FederatedAuthenticatorConfig instances.
+     * @return List of FederatedAuthenticatorListItem instances.
+     */
+    public static List<FederatedAuthenticatorListItem> build(FederatedAuthenticatorConfig[] fedAuthConfigs,
+                                                             String idpResourceId) {
+
+        List<FederatedAuthenticatorListItem> authenticators = new ArrayList<>();
+        for (FederatedAuthenticatorConfig config : fedAuthConfigs) {
+            FederatedAuthenticatorListItem authenticatorListItem = new FederatedAuthenticatorListItem();
+            authenticatorListItem.setAuthenticatorId(base64URLEncode(config.getName()));
+            authenticatorListItem.setName(config.getName());
+            authenticatorListItem.setIsEnabled(config.isEnabled());
+            authenticatorListItem.definedBy(FederatedAuthenticatorListItem.DefinedByEnum.valueOf(
+                    config.getDefinedByType().toString()));
+            String[] tags = resolveAuthenticatorTags(config);
+            if (ArrayUtils.isNotEmpty(tags)) {
+                authenticatorListItem.setTags(Arrays.asList(tags));
+            }
+            authenticatorListItem.setSelf(ContextLoader.buildURIForBody(String.format(V1_API_PATH_COMPONENT +
+                     IDP_PATH_COMPONENT + "/%s/federated-authenticators/%s", idpResourceId,
+                                    base64URLEncode(config.getName()))).toString());
```

Comment:
```suggestion
                                    base64URLEncode(config.getName()))));
```

This is not necessary right ?